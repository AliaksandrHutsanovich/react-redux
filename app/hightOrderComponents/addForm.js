import React, { useState, memo } from 'react';
import 'antd/dist/antd.css';
import { connect } from 'react-redux';
import { Input, Button } from 'antd';
import PropTypes from 'prop-types';
import { clearReDo } from '../actions/actions';
import { caseClickHandlers, kindsOfAddActions } from './utils/utils';

const addForm = (placeholder) => {
  const AddForm = (props) => {
    const { url, dispatch } = props;
    const [value, setValue] = useState('');
    const addAction = kindsOfAddActions[placeholder];

    const changeValue = (event) => {
      setValue(event.target.value);
    };

    const clickButton = () => {
      dispatch(clearReDo());
      caseClickHandlers[placeholder](dispatch, value, addAction, url);
      setValue('');
    };

    return (
      <div className="form_wrapper">
        <Input
          placeholder={placeholder}
          className="input"
          value={value}
          onChange={changeValue}
        />
        <Button
          type="primary"
          className="button"
          onClick={() => { clickButton(addAction); }}
        >
          Add
        </Button>
      </div>
    );
  };

  AddForm.propTypes = {
    dispatch: PropTypes.func.isRequired,
    url: PropTypes.string,
  };

  AddForm.defaultProps = {
    url: '',
  };

  return connect()(memo(AddForm));
};

export default addForm;
