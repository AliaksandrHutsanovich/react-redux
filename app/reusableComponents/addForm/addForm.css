.formWrapper {
    display: flex;
    position: sticky;
    top: 0px;
}

.button {
    top: 10px;
    left: 70px;
}

.input {
    display: block !important;
    width: 50% !important;
    top: 10px;
    left: 50px;
}