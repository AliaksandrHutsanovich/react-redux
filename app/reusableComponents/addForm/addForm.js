import React, { useState, memo } from 'react';
import 'antd/dist/antd.css';
import { connect } from 'react-redux';
import { Input, Button } from 'antd';
import PropTypes from 'prop-types';
import { clearReDo } from '../../actions/actions';
import { caseClickHandlers, kindsOfAddActions } from '../../hightOrderComponents/utils/utils';

import styles from './addForm.css';

const AddForm = (props) => {
  const { url, dispatch, placeholder } = props;
  const [value, setValue] = useState('');
  const addAction = kindsOfAddActions[placeholder];

  const changeValue = (event) => {
    setValue(event.target.value);
  };

  const clickButton = () => {
    dispatch(clearReDo());
    caseClickHandlers[placeholder](dispatch, value, addAction, url);
    setValue('');
  };

  return (
    <div className={styles.formWrapper}>
      <Input
        placeholder={placeholder}
        className={styles.input}
        value={value}
        onChange={changeValue}
      />
      <Button
        type="primary"
        className={styles.button}
        onClick={() => { clickButton(addAction); }}
      >
        Add
      </Button>
    </div>
  );
};

AddForm.propTypes = {
  dispatch: PropTypes.func.isRequired,
  url: PropTypes.string,
  placeholder: PropTypes.string.isRequired,
};

AddForm.defaultProps = {
  url: '',
};

export default connect()(memo(AddForm));
