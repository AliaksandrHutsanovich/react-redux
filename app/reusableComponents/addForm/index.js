export { default, AddForm } from './addForm';
