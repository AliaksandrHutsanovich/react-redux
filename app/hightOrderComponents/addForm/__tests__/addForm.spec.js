import React from 'react';
import { mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import addForm from '../addForm';

const mockStore = configureStore();
const initialState = {};
const CustomAddForm = addForm('Add new category');


describe('Unit tests of add form component', () => {
  const store = mockStore(initialState);
  const Component = mount(<CustomAddForm store={store} />);
  it('full render test', () => {
    expect(Component).toMatchSnapshot();
  });

  it('input should be changable', () => {
    Component.find('input').simulate(
      'change',
      {
        target: {
          value: 'javascript',
        },
      },
    );
    expect(Component.find('input').props().value).toEqual('javascript');
  });

  it('button should be clickable', () => {
    Component.find('button').simulate('click');
  });
});
