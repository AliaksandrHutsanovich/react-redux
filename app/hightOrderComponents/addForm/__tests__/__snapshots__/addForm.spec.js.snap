// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit tests of add form component full render test 1`] = `
<Connect(Component)
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <AddForm
    dispatch={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <div
      className="formWrapper"
    >
      <Input
        className="input"
        disabled={false}
        onChange={[Function]}
        placeholder="Add new category"
        type="text"
        value=""
      >
        <input
          className="ant-input input"
          disabled={false}
          onChange={[Function]}
          onKeyDown={[Function]}
          placeholder="Add new category"
          type="text"
          value=""
        />
      </Input>
      <Button
        block={false}
        className="button"
        ghost={false}
        loading={false}
        onClick={[Function]}
        type="primary"
      >
        <Wave>
          <button
            className="ant-btn button ant-btn-primary"
            onClick={[Function]}
            type="button"
          >
            <span
              key=".0"
            >
              Add
            </span>
          </button>
        </Wave>
      </Button>
    </div>
  </AddForm>
</Connect(Component)>
`;
